File: src/dotnet/Ansight/Pairing/Models/DeviceRuntimeProfile.cs
namespace Ansight.Pairing.Models;

public sealed class DeviceRuntimeProfile
{
    public int? Primary { get; set; }
    public string? PrimaryVersion { get; set; }
    public DeviceRuntimeEngineProfile? Engine { get; set; }
    public List<DeviceRuntimeStackEntry>? Stack { get; set; }
    public bool? AotEnabled { get; set; }
    public bool? JitEnabled { get; set; }
}

File: src/ios/Sources/AnsightCore/DeviceRuntimeProfile.swift
import Foundation

#if canImport(Darwin)
import Darwin
#endif

#if canImport(UIKit)
import UIKit
#endif

public struct DeviceRuntimeProfile: Sendable, Codable, Equatable {
    public var primary: Int?
    public var primaryVersion: String?
    public var engine: DeviceRuntimeEngineProfile?
    public var stack: [DeviceRuntimeStackEntry]?
    public var aotEnabled: Bool?
    public var jitEnabled: Bool?
}
